src/scanner.h

```diff
@@ -30,95 +30,95 @@
 */
 enum SCANNER_SYMBOLS {
-    S_ILLEGAL           = 0UL,
-
-    S_IDENT             = (1UL<< 3),
-    S_UNBIND            = (1UL<< 3)+1,
-    S_ISBOUND           = (1UL<< 3)+2,
-    S_TRYNEXT           = (1UL<< 3)+3,
-    S_INFO              = (1UL<< 3)+4,
-    S_ASSERT            = (1UL<< 3)+5,
-    S_LBRACK            = (1UL<< 4)+0,
-    S_LBRACE            = (1UL<< 4)+1,
-    S_BLBRACK           = (1UL<< 4)+2,
-    S_RBRACK            = (1UL<< 5)+0,
-    S_RBRACE            = (1UL<< 5)+1,
-    S_DOT               = (1UL<< 6)+0,
-    S_BDOT              = (1UL<< 6)+1,
-    S_LPAREN            = (1UL<< 7),
-    S_RPAREN            = (1UL<< 8),
-    S_COMMA             = (1UL<< 9)+0,
-    S_DOTDOT            = (1UL<< 9)+1,
-    S_COLON             = (1UL<< 9)+2,
-    S_READWRITE         = (1UL<< 9)+3,
-    S_READONLY          = (1UL<< 9)+4,
-    S_DOTDOTDOT         = (1UL<< 9)+5,
-
-    S_INT               = (1UL<<10)+0,
-    S_FLOAT             = (1UL<<10)+1,
-
-    S_TRUE              = (1UL<<11)+0,
-    S_FALSE             = (1UL<<11)+1,
-    S_CHAR              = (1UL<<11)+2,
-    S_STRING            = (1UL<<11)+3,
-    S_TILDE             = (1UL<<11)+4,
-    S_HELP              = (1UL<<11)+5,
-    S_PRAGMA            = (1UL<<11)+6,
-
-
-    S_REC               = (1UL<<12)+0,
-
-    S_FUNCTION          = (1UL<<13),
-    S_LOCAL             = (1UL<<14),
-    S_END               = (1UL<<15),
-    S_MAPTO             = (1UL<<16),
-
-    S_MULT              = (1UL<<17)+0,
-    S_DIV               = (1UL<<17)+1,
-    S_MOD               = (1UL<<17)+2,
-    S_POW               = (1UL<<17)+3,
-
-    S_PLUS              = (1UL<<18)+0,
-    S_MINUS             = (1UL<<18)+1,
-
-    S_EQ                = (1UL<<19)+0,
-    S_LT                = (1UL<<19)+1,
-    S_GT                = (1UL<<19)+2,
-    S_NE                = (1UL<<19)+3,
-    S_LE                = (1UL<<19)+4,
-    S_GE                = (1UL<<19)+5,
-    S_IN                = (1UL<<19)+6,
-
-    S_NOT               = (1UL<<20)+0,
-    S_AND               = (1UL<<20)+1,
-    S_OR                = (1UL<<20)+2,
-
-    S_ASSIGN            = (1UL<<21),
-
-    S_IF                = (1UL<<22)+0,
-    S_FOR               = (1UL<<22)+1,
-    S_WHILE             = (1UL<<22)+2,
-    S_REPEAT            = (1UL<<22)+3,
-    S_ATOMIC            = (1UL<<22)+4,
-
-    S_THEN              = (1UL<<23),
-    S_ELIF              = (1UL<<24)+0,
-    S_ELSE              = (1UL<<24)+1,
-    S_FI                = (1UL<<25),
-    S_DO                = (1UL<<26),
-    S_OD                = (1UL<<27),
-    S_UNTIL             = (1UL<<28),
-
-    S_BREAK             = (1UL<<29)+0,
-    S_RETURN            = (1UL<<29)+1,
-    S_QUIT              = (1UL<<29)+2,
-    S_QQUIT             = (1UL<<29)+3,
-    S_CONTINUE          = (1UL<<29)+4,
-
-    S_SEMICOLON         = (1UL<<30)+0,
-    S_DUALSEMICOLON     = (1UL<<30)+1,
-
-    S_EOF               = (1UL<<31),
+    S_ILLEGAL           = 0,
```

Comment:
I hope this is not a stupid question, but why is this equivalent to ` S_ILLEGAL = 0UL,`? In the above changes, you always replace the `UL` construction with `(UInt)`, here you just drop it.

Reply:
Not a stupid question at all! I dropped it here because all those values fit into 32bit, and the resulting enum values will be stored as 32bit, too, so there is no point in using `(UInt)1` or `1UL`.

Side remark: the C standard only guarantees that an enum must be able to represent any `int` value, which means 32bit these days -- we are thus careful not to use more than 31 bits here (we can't use the sign bit). GCC actually supports 64bit enum values, but there are C compilers which don't, and the clang static analyzer warns about it.

Reply:
Ah that is a good point, thanks!